src/Hl7.Fhir.Base/Serialization/FhirJsonPocoDeserializer.cs

```diff
@@ -713,6 +720,26 @@ FhirJsonPocoDeserializerState state
                     new(contents, ERR.STRING_ISNOTAN_INSTANT.With(ref reader, contents));
             }
 
+            static (object?, FhirJsonException?) readLong(ref Utf8JsonReader reader, Type? fhirType)
+            {
+                // convert string in json to a long.
+                var contents = reader.GetString()!;
+
+                return long.TryParse(contents, out var parsed) switch
+                {
+                    true when isInteger64() => new(parsed, null),
+                    true => new(parsed, ERR.LONG_CANNOT_BE_PARSED.With(ref reader, contents, typeName())),
+                    false when isInteger64() => new(contents, ERR.NUMBER_CANNOT_BE_PARSED.With(ref reader, contents, nameof(Integer64))),
+                    false => new(contents, ERR.LONG_CANNOT_BE_PARSED.With(ref reader, contents, typeName()))
```

Comment:
Since this is a string, but it cannot be parsed as a number, something is wrong. But the error message says: "Json number '{0}' cannot be parsed as a {1}. Json token should be string." - which is confusing since the token IS a string at this point.

Reply:
I've changed the error message. I even added a new errorcode for that.